crates/core/tedge_actors/src/servers/builders.rs

```diff
@@ -201,16 +201,6 @@ impl<S: Server + Clone> Builder<ConcurrentServerActor<S>> for ServerActorBuilder
     }
 }
 
-impl<S: Server, K> ServiceProvider<S::Request, S::Response, NoConfig> for ServerActorBuilder<S, K> {
-    fn connect_consumer(
-        &mut self,
-        config: NoConfig,
-        response_sender: DynSender<S::Response>,
-    ) -> DynSender<S::Request> {
-        self.box_builder.connect_consumer(config, response_sender)
-    }
-}
```

Comment:
Removing this implementation is the first step to deprecate the service provider and consumer traits.

Server actors have been the main motivation for the `ServiceProvider` trait. With the now deprecated original design, registering a client actor used to require an exchange of two senders (one given by the client and a second one built specifically for the client and returned by the server to the client). Now that clients send a reply-to address along their requests, the `MessageSink` trait is enough to register a client to a server.